proposals/NNNN-add-preferred-FPS.md

```diff
@@ -0,0 +1,128 @@
+# Add preferred FPS to VideoStreamingCapability
+
+* Proposal: [SDL-NNNN](NNNN-add-preferred-FPS.md)
+* Author: [Shinichi Watanabe](https://github.com/shiniwat)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Java Suite / RPC]
+
+## Introduction
+
+This proposal extends SDL video streaming feature by adding preferred FPS to VideoStreamingCapability, so that frame rate of video streaming can be adjusted per OEM.
+
+## Motivation
+
+Currently, SDL video streaming refers to VideoStreamingCapability, and mobile libraries respect for preferred resolution, maximum bitrate, video format, etc.
+However frame rate of video streaming totally depends on mobile applications, and some OEM's head unit may not work well for high frame rate, such as 60 fps.
+This proposal allows head units to specify the preferred FPS, so that mobile library can refer to the specified value. 
+
+## Proposed solution
+
+The proposed solution is to add preferred FPS to VideoStreamingCapability. On a head unit, preferred FPS can be varied on screen resolution; i.e. frame rate can be higher for lower resolution, but the preferred FPS value can be used for the case where screen resolution is the preferred resolution value.
+Mobile Proxies should respect for the preferred FPS value, as well as they do for preferred resolution, maximum bitrate, etc.
+
+## Detailed design
+
+### RPC change (Mobile_API and HMI_API)
+
+Add preferredFPS to VideoStreamingCapability struct in both APIs.
+
+```xml
+    <struct name="VideoStreamingCapability" since="4.5">
+        <description>Contains information about this system's video streaming capabilities.</description>
+        ...
+        <!-- new param -->
+        <param name="preferredFPS" type="Integer" minvalue="0" maxvalue="2147483647" mandatory="false">
+            <description>Preferred frame rate per second. Mobile application should take this value into account for capturing and encoding video frame.</description>
+        </param>
+    </struct>
+```
+
+Like other parameters in VideoStreamingCapability, preferredFPS should be returned in GetSystemCapability RPC response.
+
+### iOS mobile proxy changes
+
+Extend SDLVideoStreamingCapability class to have preferredFPS property:
+
+SDLVideoStreamingCapability.h:
+
+```objc
+/**
+ *  Preferred frame rate (assuming the case where resolution is the same as preferredResolution)
+ *  Optional
+ */
+@property (nullable, strong, nonatomic) NSNumber<SDLInt> *preferredFPS;
+
+```
+
+SDLVideoStreamingCapability.m:
+```objc
+- (void)setPreferredFPS:(nullable NSNumber<SDLInt> *)preferredFPS {
+    [store sdl_setObject:preferredFPS forName:SDLRPCParameterNamePreferredFPS];
+}
+
+- (nullable NSNumber<SDLInt> *)preferredFPS {
+    return [store sdl_objectForName:SDLRPCParameterNamePreferredFPS ofClass:NSNumber.class error:nil];
+}
+```
+
+And then, modify SDLStreamingVideoLifecycleManager#didEnterStateVideoStreamStarting, so that preferredFPS value would be taken into account:

```

Comment:
```suggestion
And then, modify SDLStreamingVideoLifecycleManager#didEnterStateVideoStreamStarting, so that `preferredFPS` value would be taken into account:
```

Reply:
Made SDLStreamingVideoLifecycleManager#didEnterStateVideoStreamStarting to code snippet as well.